### PR TITLE
[TT-4571] Fix the case where multiple securitySchemes exist

### PR DESCRIPTION
**Pull request by furkansenharputlu**
This PR fixes the case where `Security` object has multiple securitySchemes added as `AND` condition.

### CODE REVIEW

File: apidef/oas/security.go [hunk 1]
```diff
@@ -71,14 +70,18 @@ func (s *OAS) extractSecurityTo(api *apidef.APIDefinition) {
...
 		return
 	}
 
-	for name := range s.Security[0] {
-		switch s.Components.SecuritySchemes[name].Value.Type {
-		case apiKey:
-			s.extractTokenTo(api, name)
+	for schemeName := range s.getTykSecuritySchemes() {
+		if _, ok := s.Security[0][schemeName]; ok {
+			switch s.Components.SecuritySchemes[schemeName].Value.Type {
```
Comment (titpetric) [inline apidef/oas/security.go:79]: switch + single case:, perhaps convert to if? (gosimple)

**Review by titpetric — APPROVED** [review]

File: apidef/oas/security.go (hunk 1 above)
Comment (furkansenharputlu) [reply apidef/oas/security.go:79]: The switch will have more cases as we add other authentication modes.

**Review by jeffy-mathew — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3922/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1951120592)

**Pull request merged**